src/rules/rust_style.rs

```diff
@@ -0,0 +1,2070 @@
+        for prop in *props {
+          match prop {
+            ast_view::ObjectPatProp::KeyValue(ast_view::KeyValuePatProp {
+              ref key,
+              ref value,
+              ..
+            }) => match value {
+              ast_view::Pat::Ident(value_ident) => {
+                self.check_ident_snake_cased(
+                  value_ident,
+                  IdentToCheck::object_pat(
+                    &key.string_repr().unwrap_or_else(|| "[KEY]".to_string()),
+                    Some(&value_ident.id.inner),
+                    false,
+                    pat_in_var_declarator(pat.into()),
+                  ),
+                );
+              }
+              ast_view::Pat::Assign(ast_view::AssignPat {
+                left: ast_view::Pat::Ident(value_ident),
+                ..
+              }) => {
+                self.check_ident_snake_cased(
+                  value_ident,
+                  IdentToCheck::object_pat(
+                    &key.string_repr().unwrap_or_else(|| "[KEY]".to_string()),
+                    Some(&value_ident.id.inner),
+                    true,
+                    pat_in_var_declarator(pat.into()),
+                  ),
+                );
+              }
+              _ => {
+                self.check_pat(value, ctx);
+              }
+            },
+            ast_view::ObjectPatProp::Assign(ast_view::AssignPatProp {
+              ref key,
+              ref value,
+              ..
+            }) => {
+              let has_default = value.is_some();
+              let in_var_declarator = pat_in_var_declarator(pat.into());
+              if !in_var_declarator {
+                self.check_ident_snake_cased(
+                  key,
+                  IdentToCheck::object_pat::<&str, &str>(
+                    &key.inner.as_ref(),
+                    None,
+                    has_default,
+                    in_var_declarator,
+                  ),
+                );
+              }
+            }
+            ast_view::ObjectPatProp::Rest(ast_view::RestPat {
+              ref arg,
+              ..
+            }) => {
+              self.check_pat(arg, ctx);
+            }
+          }
+        }
+      }
+      ast_view::Pat::Assign(ast_view::AssignPat { ref left, .. }) => {
+        self.check_pat(left, ctx);
+      }
+      ast_view::Pat::Expr(expr) => {
+        if let ast_view::Expr::Ident(ident) = expr {
+          self.check_ident_snake_cased(
+            ident,
+            IdentToCheck::variable(ident.inner),
+          );
+        }
+      }
+      ast_view::Pat::Invalid(_) => {}
+    }
+  }
+}
+
+impl Handler for RustStyleHandler {
+  fn assign_expr(&mut self, e: &AssignExpr, _ctx: &mut Context) {
+    let idents: Vec<deno_ast::swc::ast::Ident> = find_lhs_ids(&e.left);
+
+    for ident in idents {
+      self.check_ident_snake_cased_or_screaming_snake_cased(
+        &ident.range(),
+        IdentToCheck::variable_or_constant(ident.to_id().0),
```

Comment:
?should global assign of camelCase variables be valid, because one has no control over the identifier?
if so, we should remove `fn assign_expr`.